spec/models/staff_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe Staff, type: :model do
+  subject {FactoryBot.build :staff}
+
+  describe "Associations" do
+    it { is_expected.to have_many(:news).dependent(:destroy) }
+  end
+
+  describe "image" do
+    it { expect(subject.image).to be_attached }
+  end
+
+  describe "Validations" do
+    it { is_expected.to validate_presence_of(:user_name).with_message(I18n.t("errors_blank")) }
+    it { is_expected.to validate_length_of(:user_name).is_at_most(Settings.max_user_name)}
+    it { is_expected.to validate_presence_of(:full_name).with_message(I18n.t("errors_blank")) }
+    it { is_expected.to validate_length_of(:full_name).is_at_most(Settings.max_user_name)}
+    it { is_expected.to validate_presence_of(:email).with_message(I18n.t("errors_blank")) }
+    it { is_expected.to allow_value("yen@gmail.com").for(:email) }
+    it { is_expected.to validate_uniqueness_of(:email).case_insensitive.with_message(I18n.t("errors_taken")) }
+    it { is_expected.to validate_presence_of(:password).with_message(I18n.t("errors_blank")) }
+    it { is_expected.to validate_length_of(:password).is_at_least(Settings.min_pass) }
+    it { is_expected.to validate_confirmation_of(:password) }
+  end
+
+  describe "Scopes" do
+    let!(:u1) { FactoryBot.create :staff, user_name: "yen", full_name: "yen", email: "yen@gmail.com"  }
+    let!(:u2) { FactoryBot.create :staff, user_name: "phuong", full_name: "phuong", email: "phuong@gmail.com" }
```

Comment:
let vs let! khác gì nhau?
truyền full_name vào là được rồi

Reply:
- let tạo ra object, data trong example, nó là lazy-evaluated, nghĩa là let chỉ chạy khi và chỉ khi nào được gọi đến. Nó chỉ có giá trị trong từng example, nếu gọi đến một biến let nhiều lần trong cùng một example thì nó sẽ chỉ khai báo khối let một lần duy nhất thôi (ở lần gọi đầu tiên). 
- let! thì tạo object ngay khi được chạy. Nó được tính toán và lưu lại trước khi khối it được thực thi.